### DISCUSSION TITLE
Empty Artifacts Folder

### DISCUSSION THREAD

**Discussion by lgacnik** [microsoft/vcpkg#31431] [Q&A]
I'm using a VS Code [extension](https://marketplace.visualstudio.com/items?itemName=stmicroelectronics.stm32-vscode-extension) that most probably uses `vcpkg`. After installing this extension, each time I run VS Code I get the following error:

```
> Using vcpkg executable at "c:\Users\Luka\.vscode\extensions\ms-vscode.vscode-embedded-tools-0.7.0-win32-x64\assets\platform\win32-x64\vcpkg.exe".
> Using VCPKG_ROOT at "c:\Users\Luka\AppData\Roaming\Code\User\globalStorage\ms-vscode.vscode-embedded-tools\vcpkg\root".
> Activating environment at "c:\Repository\Code\STM32Cube\TestSTM32vscode\vcpkg-configuration.json".
> Version of vcpkg used to create VCPKG_ROOT: "2023-03-14-105f7f7e8a5ea10dfac9874ff9d1a6bacba5f454".
> Running vcpkg with arguments ["version"].
vcpkg package management program version 2023-03-14-105f7f7e8a5ea10dfac9874ff9d1a6bacba5f454

See LICENSE.txt for license information.
> Bundled version of vcpkg: "2023-03-14-105f7f7e8a5ea10dfac9874ff9d1a6bacba5f454".
> Running vcpkg with arguments ["activate","--project","c:\\Repository\\Code\\STM32Cube\\TestSTM32vscode","--json","C:\\Users\\Luka\\AppData\\Local\\Temp\\0b0bfd2e9086e9804700cf10fe331514"].
warning: vcpkg-artifacts is experimental and may change at any time.
Downloading vcpkg-artifacts bundle 2023-03-14...
Downloading https://github.com/microsoft/vcpkg-tool/releases/download/2023-03-14/vcpkg-ce.tgz
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module 'C:\Users\Luka\AppData\Local\vcpkg\artifacts-2023-03-14\node_modules\'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.16.0
> Failed with exit code "1".
> Failed to activate environment at "c:\Repository\Code\STM32Cube\TestSTM32vscode\vcpkg-configuration.json".
```
I have checked the files it cannot find and found out that the `artifacts-2023-03-14` is empty. I have tried several possible solution to this problem [here](https://stackoverflow.com/questions/53545800/internal-modules-cjs-loader-js582-throw-err/54964538#54964538) but didn't manage to change anything.

Any idea or hint to possible solution is very much appreciated!

**Comment by AugP** [comment]
Tagging a few folks who may be able to help here: @benmcmorran @BillyONeal @robotdad.

Looks like vcpkg is failing somewhere when called within the VS Code Embedded Tools extension.

**Reply by BillyONeal** [reply]
3 14 is old and we've touched how this works since... try updating?

**Reply by benmcmorran** [reply]
Can you try switching to the pre-release version of Embedded Tools (0.7.230516001 as of this writing)? There should be a button in the extension marketplace to switch. The update to the bundled version of vcpkg hasn't made it to the official release channel yet while we work through [splitting it into its own extension](https://github.com/microsoft/vscode-embedded-tools/issues/51).
